WordPress/Classes/ViewRelated/Post/PostNoticeViewModel.swift

```diff
@@ -102,8 +106,23 @@ struct PostNoticeViewModel {
         case .pending:
             return NSLocalizedString("Post pending review", comment: "Title of notification displayed when a post has been successfully saved as a draft.")
         default:
-            return NSLocalizedString("Post published", comment: "Title of notification displayed when a post has been successfully published.")
+            if isFirstTimePublish(post) {
+                return NSLocalizedString("Post published", comment: "Title of notification displayed when a post has been successfully published.")
+            } else {
+                return NSLocalizedString("Post updated", comment: "Title of notification displayed when a post has been successfully updated.")
+            }
+        }
+    }
+
+    private func isFirstTimePublish(_ post: AbstractPost) -> Bool {
+        guard let publishDate = post.dateCreated, let modifiedDate = post.dateModified else {
+            return false
+        }
+        let publishDateWithThreshold = Calendar.current.date(byAdding: .second, value: 5, to: publishDate) ?? publishDate
```

Comment:
This threshold seemed necessary only for self-hosted sited

Reply:
Is this due to system clocks being slightly out of sync, or something else? Maybe we can add a comment to explain the need for the threshold, wdyt?

Reply:
In my understanding this happens because the publishDate is updated when the remote post is received. In any case I updated the implementation to avoid such workarounds.